### PR TITLE
Cats now display a variety of catlike behaviors!

### PR DESCRIPTION
**Pull request by harmonyn**
There were sprites for resting and sitting cats left unused in the DMI; they are now used. Trying to think of more cat stuff but it's a shame to have these not be used anywhere

Runtime for cutest pet :cat:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -41,6 +41,23 @@
...
+			emote("me", 1, pick("sits down."))
+			icon_state = "[icon_living]_sit"
+			resting = 1
+		else if (prob(1))
+			if (resting)
+				emote("me", 1, pick("gets up and meows."))
+				icon_state = "[icon_living]"
+				resting = 0
+			else
+				emote("me", 1, pick("grooms her fur."))
```
Comment (francinum) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: its

Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: \her

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 2]
```diff
@@ -41,6 +41,23 @@
...
+			emote("me", 1, pick("stretches out for a belly rub.", "wags its tail."))
+			icon_state = "[icon_living]_rest"
+			resting = 1
+		else if (prob(1))
+			emote("me", 1, pick("sits down."))
+			icon_state = "[icon_living]_sit"
+			resting = 1
+		else if (prob(1))
+			if (resting)
+				emote("me", 1, pick("gets up and meows."))
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: you're not picking anything, so just put the string as the argument

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 3]
```diff
@@ -41,6 +41,23 @@
 	gold_core_spawnable = 0
 
 /mob/living/simple_animal/pet/cat/Life()
+	if(!stat && !buckled)
+		if(prob(1))
+			emote("me", 1, pick("stretches out for a belly rub.", "wags its tail."))
+			icon_state = "[icon_living]_rest"
+			resting = 1
+		else if (prob(1))
+			emote("me", 1, pick("sits down."))
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: you're not picking anything, so just put the string as the argument

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 4]
```diff
@@ -41,6 +41,23 @@
 	gold_core_spawnable = 0
 
 /mob/living/simple_animal/pet/cat/Life()
+	if(!stat && !buckled)
+		if(prob(1))
+			emote("me", 1, pick("stretches out for a belly rub.", "wags its tail."))
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: cats don't really wag....swish maybe, but not wag.

Also, cats and belly rugs are two combinations that notoriously do not mix.

Comment (bgobandit) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: my cat liked belly rubs :(

Comment (GunHog) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: They love head pats, though!

Comment (Cheridan) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: You should be able to rub the belly exactly twice. On the third rub you get mauled.

Comment (as334) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: Maybe a random number from 1-4 rubs? That seems to be the deal with my cat.

Comment (bgobandit) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: nanotrasen probably declaws their cats

monsters

Comment (CosmicScientist) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: They can still bite, unless they were demouthed...

Why not "wags \her tail" /  "wags \his tail"? You put gender on grooming.

**Comment by Ultimate-Chimera** [comment]
Why not dogs? :(

**Comment by Niknakflak** [comment]
:+1: kitty!

**Comment by Shadowlight213** [comment]
@ultimate-Chimera we have ian.

**Comment by Ultimate-Chimera** [comment]
@Shadowlight213  But Ian doesn't have rest sprites. :(

**Comment by PKPenguin321** [comment]
commit names tho

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 5]
```diff
@@ -66,6 +66,23 @@
 	gold_core_spawnable = 0
 
 /mob/living/simple_animal/pet/cat/Life()
+	if(!stat && !buckled)
+		if(prob(1))
+			emote("me", 1, pick("stretches out for a belly rub.", "wags its tail."))
+			icon_state = "[icon_living]_rest"
+			resting = 1
+		else if (prob(1))
```
Comment (tkdrg) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: Conditional probability means that these get less and less likely the further down you go (1%, 0.99%, 0.9801%, etc). Granted, not that big of a deal, but probably not intended. Should probably either do a pick/prob chain or `switch(pick(rand(1, 100)))`.

**Comment by Dorsisdwarf** [comment]
Does not make cats occasionally run around at speed 0 for a few seconds, then sit down and go to sleep.

:-1:

**Comment by Niknakflak** [comment]
:+1: merge when

File: code/modules/mob/living/simple_animal/friendly/cat.dm (hunk 4 above)
Comment (duncathan) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: The macros don't work

File: code/modules/mob/living/simple_animal/friendly/cat.dm (hunk 5 above)
Comment (duncathan) [inline code/modules/mob/living/simple_animal/friendly/cat.dm]: I don't think it matters

**Pull request merged**